### ISSUE TITLE
Make abstract class with abstract methods not callable

### ISSUE THREAD

**Issue by juk-r** [microsoft/pyright#6430]
For example:

```python
from abc import ABC, abstractmethod
from typing import Callable, Any

class AbstractClass(ABC):
    @abstractmethod
    def abstract_method(self): ...
    
def accept_callable(x: Callable[..., Any]): ...

accept_callable(AbstractClass)
```
should report
```
Argument of type "type[AbstractClass]" cannot be assigned to parameter "x" of type "(...) -> Any" in function "accept_callable"
  Type "type[AbstractClass]" cannot be assigned to type "(...) -> Any"
    "AbstractClass.abstract_method" is abstract
```

**Comment by erictraut** [comment]
I'm not convinced this should generate an error. From a type compatibility perspective, there is no violation here. Mypy doesn't generate an error in this case either. 

I'm going to reject this feature request for now. I might reconsider in the future if there's sufficient signal from pyright users (e.g. in the form of upvotes on this issue).
